common/vpcclient/vpcfilevolume/get_snapshot.go

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2025 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package vpcfilevolume ...
+package vpcfilevolume
+
+import (
+	"time"
+
+	"github.com/IBM/ibmcloud-volume-file-vpc/common/vpcclient/client"
+	"github.com/IBM/ibmcloud-volume-file-vpc/common/vpcclient/models"
+	util "github.com/IBM/ibmcloud-volume-interface/lib/utils"
+	"go.uber.org/zap"
+)
+
+// GetSnapshot GETs from /shares/{share_id}/snapshots/{id}
+func (ss *SnapshotService) GetSnapshot(shareID string, snapshotID string, ctxLogger *zap.Logger) (*models.Snapshot, error) {
+	ctxLogger.Debug("Entry Backend GetSnapshot")
+	defer ctxLogger.Debug("Exit Backend GetSnapshot")
+
+	defer util.TimeTracker("GetSnapshot", time.Now())
+
+	operation := &client.Operation{
+		Name:        "GetSnapshot",
+		Method:      "GET",
+		PathPattern: snapshotIDPath,
+	}
+
+	var snapshot models.Snapshot
+	var apiErr models.Error
+
+	request := ss.client.NewRequest(operation).PathParameter(shareIDParam, shareID)
+	ctxLogger.Info("Equivalent curl command", zap.Reflect("URL", request.URL()), zap.Reflect("Operation", operation))
+
+	req := request.PathParameter(snapshotIDParam, snapshotID)
+
+	_, err := req.JSONSuccess(&snapshot).JSONError(&apiErr).Invoke()
+	if err != nil {
+		return nil, err
+	}
+
+	return &snapshot, nil
+}
+
+// GetSnapshotByName GETs /shares/{share_id}/snapshots
+func (ss *SnapshotService) GetSnapshotByName(shareID string, snapshotName string, ctxLogger *zap.Logger) (*models.Snapshot, error) {
+	ctxLogger.Debug("Entry Backend GetSnapshotByName")
+	defer ctxLogger.Debug("Exit Backend GetSnapshotByName")
+
+	defer util.TimeTracker("GetSnapshotByName", time.Now())
+
+	// Get the snapshot details for a single snapshot, ListSnapshotFilters will return only 1 snapshot in list
+	//filters := &models.LisSnapshotFilters{Name: snapshotName} //Revert this once VPC IaaS fix the issue https://ibm-iaas.atlassian.net/browse/DFS-3857
+	snapshots, err := ss.ListSnapshots(shareID, 1, "", nil, ctxLogger)
```

Comment:
why we want to hard code as 1, are you getting only one snapshot only, what about if a share has many snapshot

Reply:
This is getSnapshotByName and this should always return 1 record. That is why it is hardcoded.

Reply:
Opened GHE https://github.com/IBM/ibmcloud-volume-file-vpc/issues/85 to track this.